### PR TITLE
Fix for factory class naming

### PR DESCRIPTION
**Pull request by alextrueman**
Changing the process of name selection for factory defining

Example:
```
rails g model UserMedia filename:string --force-plural
...
invoke      factory_bot
create        spec/factories/user_media.rb
```
it generates the correct factory naming:
```ruby
FactoryBot.define do
  factory :user_media do
    filename { "MyString" }
  end
end
```

Fixes: [356](https://github.com/thoughtbot/factory_bot_rails/issues/356)

### CODE REVIEW

**Comment by composerinteralia** [comment]
Hey there, Alex! It's been a while! Thanks for the contribution.

The build hasn't been passing on master for a while. I am trying to fix that in https://github.com/thoughtbot/factory_bot_rails/pull/367 (still failing for Rails 6 at the moment).

**Comment by alextrueman** [comment]
> Hey there, Alex! It's been a while! Thanks for the contribution.
> 
> The build hasn't been passing on master for a while. I am trying to fix that in #367 (still failing for Rails 6 at the moment).

Hey there Daniel! You are right, it's been a while 😅

Yeah, I was commented your PR about this failing, I'm pretty sure we should try with rails 6.0.0 and cucumber >= 1.8.0

**Comment by alextrueman** [comment]
@composerinteralia hey there, it's green now :) May I ask you for review? 😅

File: lib/generators/factory_bot.rb [hunk 1]
```diff
@@ -14,8 +14,12 @@ def self.source_root
         File.expand_path(path)
       end
 
+      def explicit_class_name
```
Comment (composerinteralia) [inline lib/generators/factory_bot.rb]:
```suggestion
      def factory_name
```

File: lib/generators/factory_bot.rb [hunk 2]
```diff
@@ -14,8 +14,12 @@ def self.source_root
...
+        class_name.gsub("::", "").underscore
+      end
+
       def explicit_class_option
-        return if class_name == singular_table_name.camelize
+        return if class_name.underscore == explicit_class_name
```
Comment (composerinteralia) [inline lib/generators/factory_bot.rb]:
```suggestion
        return if class_name.underscore == factory_name
```

File: lib/generators/factory_bot/model/model_generator.rb [hunk 3]
```diff
@@ -54,7 +54,7 @@ def create_factory_file
 
       def factory_definition
         <<~RUBY
-            factory :#{singular_table_name}#{explicit_class_option} do
+            factory :#{explicit_class_name}#{explicit_class_option} do
```
Comment (composerinteralia) [inline lib/generators/factory_bot/model/model_generator.rb]:
```suggestion
            factory :#{factory_name}#{explicit_class_option} do
```

I think this name captures the purpose of the method a bit better. We are using it to get the name of the factory.

Comment (alextrueman) [reply lib/generators/factory_bot/model/model_generator.rb]: @composerinteralia Thank you for the suggestion! 😅

**Pull request merged**